src/compiler/watch.ts

```diff
@@ -697,20 +718,22 @@ namespace ts {
             }
         }
 
-        function removeSourceFile(path: Path) {
+        function nextSourceFileVersion(path: Path) {
             const hostSourceFile = sourceFilesCache.get(path);
             if (hostSourceFile !== undefined) {
-                if (!isString(hostSourceFile)) {
-                    hostSourceFile.fileWatcher.close();
-                    resolutionCache.invalidateResolutionOfFile(path);
+                if (isFileMissingOnHost(hostSourceFile)) {
+                    // The next version, lets set it as presence unknown file
+                    sourceFilesCache.set(path, { version: Number(hostSourceFile) + 1 });

```

Comment:
can we make it a `number` instead of a `string` since we are using it as a version?

Reply:
Done.

Reply:
so we do not need the call to `Number`?